src/core_modules/capture-core/src/components/ColumnSelector/ColumnSelector.component.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+/* eslint-disable */
+import React, { Component } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+
+import Checkbox from '@material-ui/core/Checkbox';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+
+import SettingsIcon from '@material-ui/icons/Settings';
+import ReorderIcon from '@material-ui/icons/Reorder';
+
+import i18n from '@dhis2/d2-i18n';
+
+import DragDropList from './DragDropSort/DragDropList.component';
+
+
+const styles = theme => ({
+    optionsIcon: {
+        color: theme.palette.primary.main,
+    },
+});
+
+type Props = {
+    classes: Object,
+    workingListColumnOrder: Array<Object>,
+    selectedProgramId: string,
+    onUpdateWorkinglistOrder: (workinglist: Array<Object>) => void,
+};
+
+type State = {
+    open: boolean,
+    columnList: Array<Object>,
+};
+
+class ColumnSelector extends Component<Props, State> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            open: false,
+            columnList: this.props.workingListColumnOrder,
```

Comment:
columnList is saved to state in both ColumnSelector.component and DragDropList.component. As far as I can understand the list should only be saved to state in DragDropList.component. The methods for handleToggle, handleUpdateListOrder should be moved to DragDropList.